cdap-proto/src/main/java/io/cdap/cdap/proto/security/Principal.java

```diff
@@ -37,18 +37,28 @@
   }
 
   private final String name;
+  private final String credential;
   private final PrincipalType type;
   private final String kerberosPrincipal;
   // This needs to be transient because we don't want this to be populated during deserialization since that
   // value will not be provided by the client.
   private transient int hashCode;
 
   public Principal(String name, PrincipalType type) {
-    this(name, type, null);
+    this(name, null, type, null);
   }
 
-  public Principal(String name, PrincipalType type, @Nullable String kerberosPrincipal) {
+  public Principal(String name, PrincipalType type, String kerberosPrincipal) {
+    this(name, null, type, kerberosPrincipal);
+  }
+
+  public Principal(String name, String credential, PrincipalType type) {
```

Comment:
It's better to extend parameters at the end to be consistent with existing constructors. E.g.

```
public Principal(String name, PrincipalType type, @Nullable String credential)
```

Reply:
Done